src/uct/ib/base/ib_iface.c

```diff
@@ -780,12 +769,20 @@ static int uct_ib_iface_dev_addr_is_reachable(
 
     uct_ib_address_unpack(ib_addr, &params);
 
-    if (/* at least one PKEY has to be with full membership */
-        !((params.pkey | iface->pkey) & UCT_IB_PKEY_MEMBERSHIP_MASK) ||
-        /* PKEY values have to be equal */
-        ((params.pkey ^ iface->pkey) & UCT_IB_PKEY_PARTITION_MASK)) {
-        uct_iface_populate_info_str_buf(is_reachable_params,
-                                        "none of the pkeys has full membership");
+    /* at least one PKEY has to be with full membership */
+    if (!((params.pkey | iface->pkey) & UCT_IB_PKEY_MEMBERSHIP_MASK)) {
+        UCT_IFACE_FILL_INFO_STR_BUF(
+                    is_reachable_params,
+                    "both local and remote pkeys (0x%x, 0x%x) "
+                    "have partial membership",
+                    iface->pkey, params.pkey);
+        return 0;
+    } else if (/* PKEY values have to be equal */
```

Comment:
can remove the else since have a return in line 779

Reply:
You're right